### PR TITLE
[CI][README] Update the PyTorch installation method

### PR DESCRIPTION
**Pull request by shintaro-iwasaki**
This PR fixes an error in https://github.com/pytorch/FBGEMM/actions/runs/3097186212/jobs/5013587244 by updating the PyTorch installation method.

Particularly, we use `pytorch-cuda` as recommended in the PyTorch page (https://pytorch.org/).
```sh
# Old
conda install pytorch cudatoolkit=11.3 -c pytorch
# New
conda install pytorch pytorch-cuda=11.7 -c pytorch-nightly -c nvidia
```

### CODE REVIEW

**Comment by shintaro-iwasaki** [comment]
I reasonably checked the scripts using my local GitHub FBGEMM repo, but it shouldn't be complete. Please let me test the changes at the official repository, too.

File: .github/workflows/fbgemm_nightly_build.yml [hunk 1]
```diff
@@ -5,12 +5,14 @@ name: Push Binary Nightly
...
 on:
   # For debugging, enable push/pull_request
-  # [push, pull_request]
+  pull_request:
+    branches:
+      - main
   # run every day at 10:45 AM
-  schedule:
-    - cron:  '45 10 * * *'
+  # schedule:
```
Comment (jianyuh) [inline .github/workflows/fbgemm_nightly_build.yml:12]: revert this testing changes?

**Review by jianyuh — COMMENTED** [review]
LGTM except some testing changes.

File: .github/workflows/fbgemm_nightly_build.yml (hunk 1 above)
Comment (shintaro-iwasaki) [reply .github/workflows/fbgemm_nightly_build.yml:12]: Thanks for your comment! It's still WIP. I will remove it once I can confirm this script change works on the ec2 server (currently failing)

File: fbgemm_gpu/README.md [hunk 2]
```diff
@@ -10,14 +10,14 @@ high-performance CUDA GPU operator library for GPU training and inference.
 The library provides efficient table batched embedding bag,
 data layout transformation, and quantization supports.
 
-Currently tested with CUDA 11.3, 11.5, and 11.6 in CI. In all cases, we test with PyTorch packages which are built with CUDA 11.3.
+Currently tested with CUDA 11.3, 11.5, 11.6, and 11.7 in CI. In all cases, we test with PyTorch packages which are built with CUDA 11.3.
```
Comment (mjanderson09) [inline fbgemm_gpu/README.md:13]: Should it be changed to "In all cases, we test with PyTorch packages which are built with CUDA 11.7" ?

Comment (shintaro-iwasaki) [reply fbgemm_gpu/README.md]: Thanks. Updated it to `11.7`.

File: .github/workflows/fbgemm_nightly_build.yml (hunk 1 above)
Comment (shintaro-iwasaki) [reply .github/workflows/fbgemm_nightly_build.yml]: Removed since the test finished on the ec2 server reasonably.

**Comment by shintaro-iwasaki** [comment]
This change reaches to the point where PYPI upload (https://github.com/pytorch/FBGEMM/actions/runs/3114908101), so it should be, at least, buildable on the EC2 machine we used for nightly builds. I will closely look at the result even after merging this PR.

**Pull request closed without merging**